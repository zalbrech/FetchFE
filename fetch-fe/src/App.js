import React from 'react';
import MyForm from './components/MyForm'

function App() {
  return (
    <div className="d-flex align-items-center justify-content-center wrapper">
      <MyForm />

    </div>
  )
}

export default App