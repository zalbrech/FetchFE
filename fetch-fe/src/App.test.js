import { render, screen } from '@testing-library/react';
import App from './App';

test('renders app snapshot', () => {
  // render(<App />);
  // const linkElement = screen.getByText(/learn react/i);
  // expect(linkElement).toBeInTheDocument();

  const { asFragment } = render(<App />)

  expect(asFragment(<App />)).toMatchSnapshot()
});
