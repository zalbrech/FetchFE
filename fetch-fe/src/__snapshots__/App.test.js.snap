// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app snapshot 1`] = `
<DocumentFragment>
  <div
    class="d-flex wrapper vh-100 py-4"
  >
    <div
      class="h-100 justify-content-center align-items-center form"
    >
      <div
        class="title"
      >
        Welcome
      </div>
      <div
        class="subtitle"
      >
        Please complete the form to sign up!
      </div>
      <form
        class=""
        id="theForm"
        role="theForm"
      >
        <div
          class="ic1 mb-3"
        >
          <label
            class="text-white form-label"
            for="formBasicName"
          >
            Name
          </label>
          <input
            class="form-control"
            id="formBasicName"
            name="name"
            placeholder="Enter first and last name"
            role="nameInput"
            type="name"
            value=""
          />
          <div
            class="invalid-feedback"
          >
            Please enter your name
          </div>
        </div>
        <div
          class="mb-3"
        >
          <label
            class="text-white form-label"
            for="formBasicEmail"
          >
            Email address
          </label>
          <input
            class="form-control"
            id="formBasicEmail"
            name="email"
            placeholder="Enter email"
            role="emailInput"
            type="text"
            value=""
          />
          <small
            class="text-muted form-text"
          >
            We'll never share your email with anyone else.
          </small>
          <div
            class="invalid-feedback"
          >
            Please enter a valid email
          </div>
        </div>
        <div
          class="mb-3"
        >
          <label
            class="text-white form-label"
            for="formBasicPassword"
          >
            Password
          </label>
          <input
            class="form-control"
            id="formBasicPassword"
            name="password"
            placeholder="Password"
            role="passwordInput"
            type="password"
            value=""
          />
          <small
            class="text-muted form-text"
          >
            Passowrd must be at least 6 characters long.
          </small>
          <div
            class="invalid-feedback"
          >
            Please enter a valid password
          </div>
        </div>
        <div
          class="ic2"
        >
          <label
            class="text-white form-label"
          >
            Occupation
          </label>
          <select
            class="text-center form-select"
            data-testid="occupation-test-id"
            name="occupation"
            role="occupationInput"
          >
            <option>
              -- Select Occupation --
            </option>
          </select>
          <div
            class="invalid-feedback"
          >
            Please select an occupation
          </div>
        </div>
        <div
          class="ic1"
        >
          <label
            class="text-white form-label"
            for="State"
          >
            State
          </label>
          <select
            class="text-center form-select"
            data-testid="state-test-id"
            name="state"
            role="stateInput"
          >
            <option>
              -- Select State of residence --
            </option>
            <option
              value="0"
            />
          </select>
          <div
            class="invalid-feedback"
          >
            Please enter a state
          </div>
        </div>
        <div
          class="bottom"
        >
          <button
            aria-label="formSubmitButton"
            class="submit btn btn-primary"
            name="formSubmitButton"
            role="formSubmitButton"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
